symphony/app/fbcnms-projects/workflows/src/proxy/transformers/metadata-workflowdef.js

```diff
@@ -26,40 +28,42 @@ const logger = logging.getLogger(module);
 // do not contain any prefix. Prefix is added to workflowdef if input is valid.
 function sanitizeWorkflowdefBefore(tenantId, workflowdef) {
   const tenantWithUnderscore = withUnderscore(tenantId);
-  if (workflowdef.name.indexOf('_') == -1) {
-    // validate tasks
-    for (const task of workflowdef.tasks) {
-      if (task.name.indexOf(withUnderscore(GLOBAL_PREFIX)) == 0) {
-        // noop on GLOBAL_ tasks
-      } else if (task.name.indexOf('_') == -1) {
-        // noop on unprefixed tasks
-      } else {
-        logger.error(
-          `Task name must not contain underscore: '${tenantId}'` +
-            ` in ${JSON.stringify(task)}`,
-        );
-        // TODO create Exception class
-        throw 'Task name must not contain underscore';
-      }
-    }
-    // add prefix to tasks
-    for (const task of workflowdef.tasks) {
-      // if task does not start with GLOBAL_ prefix...
-      if (task.name.indexOf(withUnderscore(GLOBAL_PREFIX)) != 0) {
-        // add tenantId prefix
-        task.name = tenantWithUnderscore + task.name;
-      }
-    }
-    // add prefix to workflow
-    workflowdef.name = tenantWithUnderscore + workflowdef.name;
-  } else {
+  if (workflowdef.name.indexOf(INFIX_SEPARATOR) > -1) {
     logger.error(
-      `Workflow name must not contain underscore: '${tenantId}'` +
-        ` in ${JSON.stringify(workflowdef)}`,
+      `Workflow name must not contain '${INFIX_SEPARATOR}': '${tenantId}'` +
+        ` in '${JSON.stringify(workflowdef)}'`,
     );
     // TODO create Exception class
     throw 'Workflow name must not contain underscore';
   }
+  // validate tasks
+  for (const task of workflowdef.tasks) {
+    if (task.name.indexOf(INFIX_SEPARATOR) > -1) {
+      logger.error(
+        `Task name must not contain '${INFIX_SEPARATOR}': '${tenantId}'` +
+          ` in '${JSON.stringify(task)}'`,
+      );
+      // TODO create Exception class
+      throw 'Task name must not contain underscore';
+    }
```

Comment:
I'm seeing repeats of this pattern where you search a string for the Infix Separator then log a message and throw an error. Could that be broken out into a separate function? I imagine it'll be useful for the entire proxy.

Reply:
Sure, I've separated a couple of util functions, this will be ongoing as those patterns emerge.